tests/framework/clients/k8s_block.go

```diff
@@ -99,52 +103,59 @@ func (rb *BlockOperation) BlockMap(name string, mountpath string) (string, error
 //Output  - k8s exec pod operation output and/or error
 func (rb *BlockOperation) BlockWrite(name string, mountpath string, data string, filename string, namespace string) (string, error) {
 	wt := "echo \"" + data + "\">" + mountpath + "/" + filename
-	writeDataToBlockPod[2] = wt
-	option := []string{name}
+	args := []string{"exec", name}
+
 	if namespace != "" {
-		option = append(option, namespace)
+		args = append(args, "-n", namespace)
 	}
-	out, err, status := rb.transportClient.Execute(writeDataToBlockPod, option)
-	if status == 0 {
-		return out, nil
+	args = append(args, "--", "sh", "-c", wt)
+
+	result, err := rb.k8sClient.Kubectl(args...)
+	if err != nil {
+		return err.Error(), fmt.Errorf("Unable to write data to pod --: %s", err)
```

Comment:
you can absolutely do

return "", fmt.Errorf(...

I have read go documentation that recommends this....

what you cannot do is

return nil, fmt.Errorf(...